### PR TITLE
Shrink published APK by ~3MB w/ABI splits

### PR DESCRIPTION
**Pull request by mikehardy**
## Purpose / Description
We include native code now and that will only continue with Rust

We must implement a 64-bit native build by Play store guidelines but we still have devices that use ARM and x86.

The only way to include 64-bit binaries and support all the users without bloating the APK as it is currently is to split via ABI

So this PR does binary ABI splits so platforms only get the binary code they need from our libraries.

That takes it from 10.5MB to 7.5MB or so

## Approach

I cribbed the style used by react-native, which I have used successfully for a couple years, and adapted the ABI-versionCode-ranges to match our versioning style

## How Has This Been Tested?

Unfortunately this is the sort of thing that you test by releasing, so this is already released as 2.12alpha7 and 2.12alpha8 to make sure it was working :fearful: 

## Learning (optional, can help others)

I put links in the comments in the code for versionCode range since that is vital to understand now and in the future.

Maximum versionCode is 2.1billion FWIW. We'll be fine.

https://github.com/facebook/react-native/blob/6d6268e903def9e4a8609e5dee6de2d2c4ab8c5e/template/android/app/build.gradle#L167-L180

## Checklist
_Please, go through these checks before submitting the PR._

- [ ] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [ ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [ ] You have performed a self-review of your own code

### CODE REVIEW

**Comment by mikehardy** [comment]
I welcome comments on this to improve it but I'm going to merge it since it does actually reflect the released state as of now. Improvements will have to be in the form of future PRs

**Comment by mikehardy** [comment]
Requested review from @david-allison-1 and @Arthur-Milchior and @timrae so everyone knows what is going on with versionCodes. Nothing about this is interesting *except* the maximum versionCode limit in Play Store, and how we manage the ranges.

**Comment by mikehardy** [comment]
Oh - one last comment, the versionCode limit (currently 2,100,000.000 looks suspiciously like a power of 2 yes? It was an int but they've already deprecated that in API28 and moved to a long internally (#5387) so I imagine we'll have 32 more bits of version code range to play with, about 40 years before we need it with this style / at our rate of version code change (which I estimate will last 50 years with no further change)

File: tools/parallel-package-release.sh [hunk 1]
```diff
@@ -34,5 +34,5 @@ for BUILD in $BUILDNAMES; do
     LCBUILD=`tr '[:upper:]' '[:lower:]' <<< $BUILD`
     ./tools/parallel-package-name.sh com.ichi2.anki.$LCBUILD AnkiDroid.$BUILD
     ./gradlew assembleRelease
-    cp AnkiDroid/build/outputs/apk/release/AnkiDroid-release.apk ../AnkiDroid-$TAG.parallel.$BUILD.apk
+    cp AnkiDroid/build/outputs/apk/release/AnkiDroid-armeabi-v7a-release.apk ../AnkiDroid-$TAG.parallel.$BUILD.apk
```
Comment (david-allison-1) [inline tools/parallel-package-release.sh:37]: Is it worth putting the fat apk for public download?

File: AnkiDroid/build.gradle [hunk 2]
```diff
@@ -66,6 +91,49 @@ android {
         }
     }
 
+    /**
+     * Set this to true to create five separate APKs instead of one:
+     *   - 2 APKs that only work on ARM/ARM64 devices
+     *   - 2 APKs that only works on x86/x86_64 devices
+     *   - a universal APK that works on all devices
```
Comment (david-allison-1) [inline AnkiDroid/build.gradle:98]: `universalApk` seems to be set to false, where is it generated?

**Comment by mikehardy** [comment]
We will never go up to alpha73 again >:-) and also it would have overflowed with the previous style anyway, it was always a hard limit.

I also think a universal APK would be nice and it can be toggled on but must not be done prior to attempting to upload to google, as google sees it as a "perfectly shadowed" APK and thus rejects the upload if it exists (I tried!).

So I think what I'll do as an optimization for the future is to add a toggle to the build that says "generate universal APK", and *after* the upload I will generate it, and use that to post to github / consume in the parallel release build.

In fact, I can generate the parallel releases automatically now and post those to the github releases as well - for public releases I think I'll do that.

I'll log all this as a separate issue - I mostly posted PR this immediately + merged it because semi-accidentally while testing it actually published the alpha. I already had the versionCode ranges nailed, so no real harm done but I wanted master state to match released state as a hard requirement for transparency + reproducibility in the community.

**Comment by timrae** [comment]
Alpha 73 was definitely an anomaly!

**Pull request merged**